### PR TITLE
[gcp] combine Shielded VMs & Confidential VMs

### PR DESCRIPTION
**Pull request by jianli-wei**
1. remove `gcp-ipi-confidential-computing-p2-f14` from _periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly_ and _periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly_
2. remove `gcp-ipi-secureboot-p2-f14` from _periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly_ and _periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly_
3. add `gcp-ipi-secureboot-confidential-computing-p2-f14`, **which combines the above 2**, to _periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly_ and _periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly_
4. note that `gcp-ipi-confidential-computing-p2-f28-destructive` and `gcp-ipi-secureboot-p2-f28-destructive` stay unchanged, which cover a single scenario each

### CODE REVIEW

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-gcp

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-gcp

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-gcp

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-ipi-security-enhancements-p2-f14

**Comment by jianli-wei** [comment]
FYI Both failures were about `cucushift-e2e-report` telling non-zero failures in cucushift-e2e, which in turn thinks itself succeeded. Anyway, the installation of them does succeed, so I think the failures had nothing to do with my changes, thanks! 

Test name | Commit | Details | Comments
-- | -- | -- | --
ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-ipi-security-enhancements-p2-f14 | b9ffc5a | [link](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/39604/rehearse-39604-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-ipi-security-enhancements-p2-f14/1660926901352402944) | INFO[2023-05-23T09:21:51Z] Step phase pre succeeded after 46m16s.
ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-gcp-ipi-security-enhancements-p2-f14 | b9ffc5a | [link](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/39604/rehearse-39604-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-gcp-ipi-security-enhancements-p2-f14/1660935644374896640) | INFO[2023-05-23T09:46:55Z] Step phase pre succeeded after 36m35s.

P.S. I'm not sure why 4.14 installation takes 10 more minutes than 4.13.

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/security-enhancements/cucushift-installer-rehearse-gcp-ipi-security-enhancements-workflow.metadata.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+	"path": "cucushift/installer/rehearse/gcp/ipi/security-enhancements/cucushift-installer-rehearse-gcp-ipi-security-enhancements-workflow.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/security-enhancements/cucushift-installer-rehearse-gcp-ipi-security-enhancements-workflow.metadata.json:2]: I suggest to use `confidential` and `secureboot` combination keyword to replace `security-enhancement` to easily figure out what features are enabled in the workflow, so I suggest to move the workflow to `cucushift/installer/rehearse/gcp/ipi/secureboot/confidential-computing` or `cucushift/installer/rehearse/gcp/ipi/confidential-computing/secureboot` for the combination.

**Comment by jianlinliu** [comment]
cc @liangxia to review.

**Pull request merged**